src/hooks/__tests__/useAlerts.test.ts

```diff
@@ -0,0 +1,83 @@
+import { act, renderHook } from '@testing-library/react';
+import { useAlerts } from '../useAlerts';
+
+describe('useAlerts', () => {
+  it('should add an alert and verify it is in the alerts list', () => {
+    const { result } = renderHook(() => useAlerts());
+
+    act(() => {
+      result.current.addAlert({ title: 'Test Alert', variant: 'success', id: 'alert-1' });
+    });
+
+    expect(result.current.alerts).toMatchSnapshot();
+  });
+
+  it('should add multiple alerts and verify all are present in the alerts list', () => {
+    const { result } = renderHook(() => useAlerts());
+    const alertCount = 3;
+
+    act(() => {
+      for (let i = 1; i <= alertCount; i++) {
+        result.current.addAlert({
+          id: `alert-${i}`,
+          title: `Test Alert ${i}`,
+          variant: i === 2 ? 'danger' : 'success'
+        });
+      }
```

Comment:
optional: (i highly recommend this)
you highlighted a use case here... the ability to apply multiple alerts at once. it might be a nice addition to simply allow the application of bulk messages without having to cycle through a listing, especially since it'd be an easier add here than in other areas of code
```
const addAlert = (options: AlertProps | AlertProps[]) => {
    setAlerts(prevAlerts => [...prevAlerts, ...((Array.isArray(options) && options) || [options])]);
  };
```

optional:
this for loop format isn't wrong, and it works, but it does come across as another typescript inflicted format... this can hide simplifications

```
  const { result } = renderHook(() => useAlerts());
    const alerts: AlertProps[] = [
      { id: 'lorem', title: 'ipsum', variant: 'success' },
      { id: 'hello', title: 'world', variant: 'danger' },
      { id: 'dolor', title: 'sit', variant: 'success' }
    ];

    act(() => {
      alerts.forEach(alert => result.current.addAlert(alert));
    });

  expect(result.current.alerts).toEqual(alerts);
```

and if you decided to also do the optional work to allow lists in addAlerts
```
  const { result } = renderHook(() => useAlerts());
    const alerts: AlertProps[] = [
      { id: 'lorem', title: 'ipsum', variant: 'success' },
      { id: 'hello', title: 'world', variant: 'danger' },
      { id: 'dolor', title: 'sit', variant: 'success' }
    ];

    act(() => {
      result.current.addAlert(alerts);
    });

    expect(result.current.alerts).toEqual(alerts);
```